eval.go

```diff
@@ -237,14 +237,24 @@ nextFileMatch:
 		}
 
 		for ; nextDoc < docCount; nextDoc++ {
+			repoID := d.repos[nextDoc]
+			repoMetadata := &d.repoMetaData[repoID]
+
 			// Skip tombstoned docs
-			if d.repoMetaData[d.repos[nextDoc]].Tombstone {
+			if repoMetadata.Tombstone {
 				continue
 			}
 
+			if opts.EvaluateFileTombstones && len(repoMetadata.FileTombstones) > 0 {
```

Comment:
minor: add a comment that this is for document tombstones. Maybe leave in a TODO saying this design is doing a lot of extra work due to needing to fetch filenames and allocate strings.

Reply:
``` 
TODO: This FileTombstones implementation (looking up by filenames) creates a lot of small allocations
			// (string filenames) and can have poor cache performance. This should be addressed before we officially
			// roll this out.
```